tests/llm/utils/tool_suggestions_config.py

```
@@ -0,0 +1,622 @@
+        end = matches[idx + 1].start() if idx + 1 < len(matches) else len(body)
+        section = body[start:end]
+
+        def _extract(label: str) -> str:
+            # Pulls the text between `**<label>:**` and the next `**...**:`
+            # bold-label or end of section. Tolerates inline-value style
+            # ("**When to use:** text") and block-value style.
+            pat = re.compile(
+                rf"\*\*{re.escape(label)}:?\*\*\s*(.*?)(?=\n\*\*[^*]+:?\*\*|\Z)",
+                re.DOTALL,
+            )
+            mm = pat.search(section)
```

Ruff (0.15.15)

[warning] Function definition does not bind loop variable `section`

(B023)